### PR TITLE
Wasmer interface refactor, part 2: `executorinterface` package redesign

### PR DESCRIPTION
**Pull request by andrei-marinica**

### CODE REVIEW

File: arwen/contexts/validator.go [hunk 1]
```diff
@@ -6,7 +6,7 @@ import (
 
 	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
 	"github.com/ElrondNetwork/wasm-vm/arwen"
-	"github.com/ElrondNetwork/wasm-vm/wasmer"
+	"github.com/ElrondNetwork/wasm-vm/executorinterface"
```
Comment (sasurobert) [inline arwen/contexts/validator.go:9]: "executor" keep it simple.

File: executorinterface/executorInstanceBuilder.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package executorinterface
+
+type CompilationOptions struct {
```
Comment (sasurobert) [inline executorinterface/executorInstanceBuilder.go:3]: add comment

File: executor/executorInstanceBuilder.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package executor
```
Comment (camilbancioiu) [inline executor/executorInstanceBuilder.go:1]: Consider renaming the file to `instanceBuilder.go`. It's already in the package `executor`.

**Review by camilbancioiu — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

File: executor/executorInstanceBuilder.go (hunk 3 above)
Comment (andrei-marinica) [reply executor/executorInstanceBuilder.go:1]: I prefixed them all, it can be sometimes hard otherwise to seek for the file in the IDE

**Review by camilbancioiu — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**